lib/querystring.js

```diff
@@ -18,7 +18,7 @@ QueryString.unescapeBuffer = function(s, decodeSpaces) {
   var n, m, hexchar;
 
   for (var inIndex = 0, outIndex = 0; inIndex <= s.length; inIndex++) {
-    var c = s.charCodeAt(inIndex);
+    var c = inIndex < s.length ? s.charCodeAt(inIndex) : NaN;
     switch (state) {
       case 'CHAR':
         switch (c) {
```

Comment:
I suspect you can eke out some more performance if you replace the calls to `charCode()` with their number literal equivalents.